numpy/lib/stride_tricks.py

```diff
@@ -105,6 +119,21 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
         # This should only happen if x.dtype is [('', 'Vx')]
         array.dtype = x.dtype
 
+    # check under/over reaching head/tail address
+    if check_bounds:
+        x_low, x_high = np.byte_bounds(x)
+        a_low, a_high = np.byte_bounds(array)
+        if x.size == 0 or a_low < x_low or x_high < a_high:
```

Comment:
What's the `size == 0` check here for? Isn't this ok providing that `a.size == 0` as well, and don't the next two conditions cover that?